src/include/pcp/pmapi.h

```diff
@@ -1199,6 +1199,7 @@ PCP_CALL extern int pmExtendFetchGroup_event(pmFG, const char *, const char *,
 			unsigned int, unsigned int *, int *);
 PCP_CALL extern int pmExtendFetchGroup_timestamp(pmFG, struct timeval *);
 PCP_CALL extern int pmFetchGroup(pmFG);
+PCP_CALL extern int pmFetchGroup_p(pmFG, int);
```

Comment:
@myllynen I think it will be cleaner to not extend the API for special cases like this (else we end up with a bunch of APIs when next special case arrives, then the next, ...), but rather to use the same mechanism that special fetch handling for counter wrap is implemented using.

Have a look at the 'wrap' field in struct __pmFetchGroup - we could so similar here: add a 'preserve' field there, check an env var (once) on context creation, use the new boolean field internally, and not need an additional API.